openhgnn/models/MAGNN.py

```diff
@@ -29,14 +25,16 @@ def build_model_from_args(cls, args, hg):
             edge_type_list = ['A-M', 'M-A', 'D-M', 'M-D']
             # in_feats: {'n1type': n1_dim, 'n2type', n2_dim, ...}
             in_feats = {'M': 3066, 'D': 2081, 'A': 5257}
+            metapath_idx_dict = mp_instance_sampler(hg, metapath_list, 'imdb4MAGNN')
 
         elif args.dataset == 'dblp4MAGNN':
             # build model
             metapath_list = ['APA', 'APTPA', 'APVPA']
             edge_type_list = ['A-P', 'P-A', 'P-T', 'T-P', 'P-V', 'V-P']
             # in_feats: {'n1type': n1_dim, 'n2type', n2_dim, ...}
             in_feats = {'A': 334, 'P': 14328, 'T': 7723, 'V': 20}
-        mp_instances = mp_instance_sampler(hg, metapath_list)
+            metapath_idx_dict = mp_instance_sampler(hg, metapath_list, 'dblp4MAGNN')
```

Comment:
You can just write
```python
metapath_idx_dict = mp_instance_sampler(hg, metapath_list, args.dataset)
```

And actually I prefer making the third argument of `mp_instance_sampler` an actual path to allow the flexibility of specifying a customizable path (see below).

Reply:
OK. I'll fix the problems related to `counting the input features on-the-fly`, `making the third argument of mp_instance_sampler an actual path` and `allow customized metapath and edge type list as arguments` in next PR. Because we'd like to implement a new trainerflow of MAGNN, and these functions are related to the implementation of MAGNN trainerflow.